core/txpool/legacypool/legacypool.go

```diff
@@ -568,9 +572,19 @@ func (pool *LegacyPool) Pending(filter txpool.PendingFilter) map[common.Address]
 				}
 			}
 		}
+		if filter.MaxDATxSize != nil && !pool.locals.contains(addr) {
+			for i, tx := range txs {
+				if types.EstimatedL1SizeScaled(tx.RollupCostData()).Cmp(maxDASizeScaled) > 0 {
+					txs = txs[:i]
```

Comment:
is capping the list valid here? i feel like we should filter rather than `break`?

Reply:
Need to cap, otherwise you'll have skipped nonce.

Reply:
Ah, this is in the context of a single address 👍